src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final float INCREASE_FACTOR = 1.5f;
+
+    private int size;
+    private int capacity = DEFAULT_CAPACITY;
+    private Object[] list;
+
+    public ArrayList() {
+        this.list = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        validateIndexForAdding(index);
+        if (size >= capacity) {
+            increaseCapacity();
+        }
+        if (index < size) {
+            shiftAllLeftFromIndex(index);
+        }
+        list[index] = value;
+        size++;
     }
```

Comment:
we can use one if here

Reply:
Honestly, I do not know how :) Could you show it?